internal/opa_inspect/opa_inspect.go

```diff
@@ -0,0 +1,151 @@
+// Copyright 2022 Red Hat, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// Functions for inspecting rego code and annotations
+package opa_inspect
+
+import (
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/open-policy-agent/opa/ast"
+)
+
+func inspectSingle(path, module string) ([]*ast.AnnotationsRef, error) {
+	mod, err := ast.ParseModuleWithOpts(path, module, ast.ParserOptions{ProcessAnnotation: true})
+	if err != nil {
+		return nil, err
+	}
+
+	as, x := ast.BuildAnnotationSet([]*ast.Module{mod})
+	if len(x) > 0 {
+		return nil, err
+	}
```

Comment:
I wrote this, now looking at it it looks dodgy, `err` is `nil` here, `x` should be named `errs`, this should be (from the top of my head), something like:

```suggestion
	as, errs := ast.BuildAnnotationSet([]*ast.Module{mod})
	if len(errs) > 0 {
		return nil, errors.New(errs.Error())
	}
```

Reply:
Fixed.